contrib/hadoop-store-builder/src/java/voldemort/store/readonly/fetcher/HdfsFetcherAggStats.java

```diff
@@ -0,0 +1,183 @@
+package voldemort.store.readonly.fetcher;
+
+import io.tehuti.Metric;
+import io.tehuti.metrics.MetricConfig;
+import io.tehuti.metrics.MetricsRepository;
+import io.tehuti.metrics.Sensor;
+import io.tehuti.metrics.stats.Rate;
+import voldemort.annotations.jmx.JmxGetter;
+import voldemort.utils.JmxUtils;
+
+import java.util.concurrent.TimeUnit;
+
+/*
+ * This class is used to capture the aggregated metrics for all the file pulls from Hdfs,
+ * and expose them through JMX Bean.
+ */
+public class HdfsFetcherAggStats {
+    // A singleton for all the data pull tasks
+    private static final HdfsFetcherAggStats stats;
+
+    // Total bytes fetched by all the data pull tasks
+    private long totalBytesFetched;
+    // Total Hdfs fetch retry number
+    private long totalFetchRetries;
+    // Total checksum failures
+    private long totalCheckSumFailures;
+    // Total file read failures
+    private long totalFileReadFailures;
+    // Total authentication failures
+    private long totalAuthenticationFailures;
+    // Total file not found failures
+    private long totalFileNotFoundFailures;
+    // Total quota exceed failures
+    private long totalQuotaExceedFailures;
+    // Total unauthorized store failures
+    private long totalUnauthorizedStoreFailures;
+
+    // Parallel active fetch number
+    // This metric is only used to capture the active fetches number right now
+    private int parallelFetches;
+    // Total Hdfs fetch number
+    private long totalFetches;
+
+    // TODO: Max number of parallel fetches (if there are more than one running at the same time)
+    // Need to use Tehuti lib
+
+    // Total number for incomplete fetches
+    private long totalIncompleteFetches;
+
+    // TODO: File fetcher average and max connection times (NameNode and DataNode)
+    // Not sure how to extract those metrics
+
+    // Metrics repository for data transfer rate
+    private static final MetricsRepository metricsRepository;
+    private static final Sensor dataTransferRateSensor;
+    private static final Metric dataTransferMetric;
+
+    // Register the global aggregated stats object in Bean Server
+    static {
+        stats = new HdfsFetcherAggStats();
+        JmxUtils.registerMbean("hdfs-fetcher-agg-stats", stats);
```

Comment:
Actually JmxUtils is handling the naming problem, and it will extract the package name and the one passed as param to compose the bean name.